Failure to build win64 target for Carla on Archlinux

mingw-w64-gcc 11.2.0-1

Running from PawPaw directory:
```sh
./bootstrap-carla.sh win64
```
ends at some point with the following error:
```
Creating qmake...
In file included from /home/redtide/PawPawBuilds/builds/win64/qtbase-opensource-src-5.9.8/src/corelib/tools/qbytearray.cpp:42:
/home/redtide/PawPawBuilds/builds/win64/qtbase-opensource-src-5.9.8/src/corelib/tools/qbytearraymatcher.h: In static member function ‘static QStaticByteArrayMatcherBase::Skiptable QStaticByteArrayMatcherBase::generate(const char*, uint)’:
/home/redtide/PawPawBuilds/builds/win64/qtbase-opensource-src-5.9.8/src/corelib/tools/qbytearraymatcher.h:103:38: error: ‘numeric_limits’ is not a member of ‘std’
  103 |         const auto uchar_max = (std::numeric_limits<uchar>::max)();
      |                                      ^~~~~~~~~~~~~~
/home/redtide/PawPawBuilds/builds/win64/qtbase-opensource-src-5.9.8/src/corelib/tools/qbytearraymatcher.h:103:58: error: expected primary-expression before ‘>’ token
  103 |         const auto uchar_max = (std::numeric_limits<uchar>::max)();
      |                                                          ^
/home/redtide/PawPawBuilds/builds/win64/qtbase-opensource-src-5.9.8/src/corelib/tools/qbytearraymatcher.h:103:61: error: ‘::max’ has not been declared; did you mean ‘std::max’?
  103 |         const auto uchar_max = (std::numeric_limits<uchar>::max)();
      |                                                             ^~~
      |                                                             std::max
In file included from /usr/include/c++/11.1.0/algorithm:62,
                 from /home/redtide/PawPawBuilds/builds/win64/qtbase-opensource-src-5.9.8/include/QtCore/../../src/corelib/global/qglobal.h:109,
                 from /home/redtide/PawPawBuilds/builds/win64/qtbase-opensource-src-5.9.8/include/QtCore/qglobal.h:1,
                 from /home/redtide/PawPawBuilds/builds/win64/qtbase-opensource-src-5.9.8/include/QtCore/../../src/corelib/thread/qatomic.h:41,
                 from /home/redtide/PawPawBuilds/builds/win64/qtbase-opensource-src-5.9.8/include/QtCore/qatomic.h:1,
                 from /home/redtide/PawPawBuilds/builds/win64/qtbase-opensource-src-5.9.8/include/QtCore/../../src/corelib/tools/qrefcount.h:43,
                 from /home/redtide/PawPawBuilds/builds/win64/qtbase-opensource-src-5.9.8/include/QtCore/qrefcount.h:1,
                 from /home/redtide/PawPawBuilds/builds/win64/qtbase-opensource-src-5.9.8/src/corelib/tools/qbytearray.h:44,
                 from /home/redtide/PawPawBuilds/builds/win64/qtbase-opensource-src-5.9.8/src/corelib/tools/qbytearray.cpp:41:
/usr/include/c++/11.1.0/bits/stl_algo.h:3467:5: note: ‘std::max’ declared here
 3467 |     max(initializer_list<_Tp> __l, _Compare __comp)
      |     ^~~
make: *** [Makefile:304: qbytearray.o] Errore 1
```

--- a/patches/qtbase/win32/02_std-numeric-limits.patch
+++ b/patches/qtbase/win32/02_std-numeric-limits.patch
@@ -1,0 +1,12 @@
+diff --git a/src/corelib/global/qglobal.h b/src/corelib/global/qglobal.h
+index 35c35ac6..38c08ccb 100644
+--- a/src/corelib/global/qglobal.h
++++ b/src/corelib/global/qglobal.h
+@@ -45,6 +45,7 @@
+ #  include <type_traits>
+ #  include <cstddef>
+ #  include <utility>
++#  include <limits>
+ #endif
+ 
+ #include <stddef.h>

